roles/lib_openshift/src/test/unit/conftest.py

```diff
@@ -0,0 +1,55 @@
+import inspect
+
+import pytest
+
+from lib_openshift.library import *  # noqa: F401,F403
+
+
+def pytest_generate_tests(metafunc):
+    if 'binary_lookup_module' in metafunc.fixturenames:
+        lookup_modules = []
+        ids = []
+        for key in list(globals().keys()):
+            value = globals()[key]
+            if inspect.ismodule(value):
+                for func, _ in inspect.getmembers(value, inspect.isfunction):
+                    if func == 'locate_oc_binary':
+                        lookup_modules.append(value)
+                        ids.append(key)
+        metafunc.parametrize('binary_lookup_module', lookup_modules, ids=ids)
+
+
+@pytest.fixture
+def mock_tmpfile_copy(request, mocker):
+    module_under_test = request.module.MODULE_UNDER_TEST
+    patch_method = module_under_test.__name__ + '.Utils.create_tmpfile_copy'
+    mocker.patch(patch_method, return_value='/tmp/mocked_kubeconfig')
+    yield
+
+
+@pytest.fixture
+def mock_locate_binary(request, mocker):
+    module_name = request.module.MODULE_UNDER_TEST.__name__
+    patch_method = module_name + '.locate_oc_binary'
+    mocker.patch(patch_method, return_value='oc')
+    yield
+
+
+@pytest.fixture
+def mock_run_cmd(request, mocker, mock_locate_binary, mock_tmpfile_copy):
```

Comment:
I tried to standardize on mocking the _run method, since it simply passes in provided arguments to subprocess and returning the value.

We could instead mock/monkeypatch the subprocess calls for the modules and then it would get us out of mocking methods we've written. If we go that route, then we could return back mocked data based on the call coming in instead of the order of the call.

I think you are right about getting too clever with trying to put this in contest, my main goal was to get standardization on mocking at a lower level than the openshift_cmd method.